### PR TITLE
Relative Date in Formbuilder updated by offset field value

### PR DESCRIPTION
**Pull request by cfmitrah**

### CODE REVIEW

File: system/handlers/formbuilder/item-types/Date.cfc [hunk 1]
```diff
@@ -13,13 +13,15 @@ component {
 
 			switch( args.relativeOperator ) {
 				case "lt":
-					args.maxDate = DateAdd( 'd', -1, Now() );
+					var diff = val(args.offset) ? val(-args.offset) : -1;
```
Comment (DominicWatson) [inline system/handlers/formbuilder/item-types/Date.cfc:16]: I'm not sure of the default `-1` here (same for the "gt" offset). I would have thought:

```
if ( Val( args.offset ) ) {
    args.maxDate = DateAdd( 'd', Val( -args.offset ), Now() );
}
```

Can you check this?

File: system/i18n/formbuilder/item-types/date.properties [hunk 2]
```diff
@@ -25,4 +25,7 @@ fieldset.relative.title=Relative dates
 relativeOperator.lt=Less than...
 relativeOperator.lte=Less than or equal to...
 relativeOperator.gt=Greater than...
-relativeOperator.gte=Greater than or equal to...
\ No newline at end of file
+relativeOperator.gte=Greater than or equal to...
+
+field.offset.title=Offset(Days)
```
Comment (DominicWatson) [inline system/i18n/formbuilder/item-types/date.properties]: Should have space between `Offset` and `(Days)`. So `Offset (days)`. Help should be:

> For relative dates, supply an optional offset to allow date restrictions such as 14 days greater than today.

**Comment by DominicWatson** [comment]
Thanks for this :) Please check review comments above.

File: system/handlers/formbuilder/item-types/Date.cfc (hunk 1 above)
Comment (cfmitrah) [inline system/handlers/formbuilder/item-types/Date.cfc:16]: They already set default value -1 in 'lt' (same for the 'gt'):
`args.maxDate = DateAdd( 'd', -1, Now() );`
I've replace this line with
`var diff = val(args.offset) ? val(-args.offset) : -1;`

Comment (DominicWatson) [inline system/handlers/formbuilder/item-types/Date.cfc:16]: Ah yes, I'm being slow. Thanks for the clarification :)

**Comment by DominicWatson** [comment]
Frantastic, thanks for this. This has been manually merged into 10.6.0

**Pull request closed without merging**